src/modules/lv2/factory.c

```diff
@@ -0,0 +1,261 @@
+/*
+ * factory.c -- the factory method interfaces
+ * Copyright (C) 2003-2022 Meltytech, LLC
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+#include <framework/mlt.h>
+
+#include <float.h>
+#include <limits.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include <lv2.h>
+
+/* lv2 extenstions */
+#include <lv2/atom/atom.h>
+#include <lv2/midi/midi.h>
+#include <lv2/port-groups/port-groups.h>
+#include <lv2/port-props/port-props.h>
+#include <lv2/presets/presets.h>
+#include <lv2/resize-port/resize-port.h>
+#include <lv2/ui/ui.h>
+#include <lv2/worker/worker.h>
+
+#include <lilv/lilv.h>
+
+#ifdef GPL
```

Comment:
Please avoid a multiple license situation. The code you borrowed for this had to for a special reason.